homeassistant/components/homekit/type_cameras.py

```diff
@@ -0,0 +1,250 @@
+        if stream_source:
+            return stream_source
+        try:
+            stream_source = asyncio.run_coroutine_threadsafe(
+                camera.stream_source(), self.hass.loop
+            ).result(10)
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.error("Failed to get stream source: %s", err)
+        return stream_source
+
+    async def start_stream(self, session_info, stream_config):
+        """Start a new stream with the given configuration."""
+        _LOGGER.debug(
+            "[%s] Starting stream with the following parameters: %s",
+            session_info["id"],
+            stream_config,
+        )
+
+        input_source = self._get_stream_source()
+        if not input_source:
+            _LOGGER.error("Camera has no stream source")
+            return False
+
+        if "-i " not in input_source:
+            input_source = "-i " + input_source
+
+        output_vars = stream_config.copy()
+        output_vars.update(
+            {
+                "v_profile": VIDEO_PROFILE_NAMES[
+                    int.from_bytes(stream_config["v_profile_id"], byteorder="big")
+                ],
+                "v_bufsize": stream_config["v_max_bitrate"] * 2,
+                "v_map": self.config[CONF_VIDEO_MAP],
+                "v_pkt_size": self.config[CONF_VIDEO_PACKET_SIZE],
+                "a_bufsize": stream_config["a_max_bitrate"] * 2,
+                "a_map": self.config[CONF_AUDIO_MAP],
+                "a_pkt_size": self.config[CONF_AUDIO_PACKET_SIZE],
+                "a_encoder": (
+                    AUDIO_ENCODER_OPUS
+                    if stream_config["a_codec"] == AUDIO_CODEC_TYPES["OPUS"]
+                    else AUDIO_ENCODER_AAC
+                ),
+            }
+        )
+        output = VIDEO_OUTPUT.format(**output_vars)
+        if self.config[CONF_SUPPORT_AUDIO]:
+            output = output + " " + AUDIO_OUTPUT.format(**output_vars)
+
+        stream = HAFFmpeg(self._ffmpeg.binary, loop=self.driver.loop)
+        opened = await stream.open(
+            cmd=[], input_source=input_source, output=output, stdout_pipe=False
+        )
+        if not opened:
+            _LOGGER.error("Failed to open ffmpeg stream")
+            return False
+
+        session_info["stream"] = stream
+        _LOGGER.info(
+            "[%s] Started stream process - PID %d",
+            session_info["id"],
+            stream.process.pid,
+        )
+        return True
+
+    async def stop_stream(self, session_info):
+        """Stop the stream for the given ``session_id``."""
+        session_id = session_info["id"]
+        stream = session_info.get("stream")
+        if stream:
+            _LOGGER.info("[%s] Stopping stream.", session_id)
+            await stream.close()
+            _LOGGER.debug("Stream process stopped.")
+        else:
+            _LOGGER.warning("No stream for session ID %s", session_id)
```

Comment:
Personally wouldn't even log at all for this case and would just make stop_stream idempotent. Unless you disagree?

Reply:
I feel like excessive logging has helped me a lot in debugging various issues since HomeKit has so many moving parts. Maybe turn this warning into debug and leave it in at least until a good number of people have gotten cameras up and running in the wild?

Reply:
Sure that works for me